ark/phenotyping/som_utils.py

```diff
@@ -52,22 +52,19 @@ def compute_cluster_avg(fovs, channels, base_dir,
         cluster_avgs = pd.concat([cluster_avgs, agg_results])
 
     # sum the counts and the channel sums
-    sum_count_totals = cluster_avgs.groupby('cluster')[channels + ['count']].sum().reset_index()
+    sum_count_totals = cluster_avgs.groupby(cluster_col)[channels + ['count']].sum().reset_index()
 
     # now compute the means using the count column
     sum_count_totals[channels] = sum_count_totals[channels].div(sum_count_totals['count'], axis=0)
 
     # drop the count column
     sum_count_totals = sum_count_totals.drop('count', axis=1)
 
-    # save the DataFrame
-    feather.write_dataframe(sum_count_totals,
-                            os.path.join(base_dir, cluster_avg_name),
-                            compression='uncompressed')
+    return sum_count_totals
 
 
 def create_fov_pixel_data(fov, channels, img_data, seg_labels,
-                          blur_factor=2, subset_proportion=0.1, seed=42):
+                          blur_factor=2, subset_proportion=0.1, seed=None):
```

Comment:
If I leave a comment about specific changes to make to multiple distinct functions in a given PR, you can assume that comment applies to the rest of the functions. For example, updating the docstrings. Having incorrect docstrings is worse than having no docstrings at all, like the mis-specified outputs for this one.

I think it would be good coding practice to start doing a self review of your PRs before tagging me. You can go through the files listed as changed and pick off the obvious stuff before I get to it.

Reply:
Can you clarify which part of the output for `create_fov_pixel_data` is mis-specified in the docstring for `create_fov_pixel_data`? I've just updated the return values to be more descriptive of the specific preprocessing going on and fixed `pd.DataFrame` to `pandas.DataFrame`. Not sure if we were also following a specific format for a `tuple` return type or not.

There was a return value missing for `compute_cluster_avg` (also fixed), did you by chance mean that function?

Reply:
Yup, it was the return value that I noticed, couldn't comment in the exact right spot bc it hadn't been changed in this PR.